system/jlib/jsocket.cpp

```diff
@@ -2116,26 +2122,32 @@ bool CSocket::check_connection()
     if (state != ss_open) 
         return false;
 
-    unsigned retrycount=100;
-EintrRetry:
-    int rc;
-    if (sockmode==sm_udp_server) { // udp server
-        DEFINE_SOCKADDR(u);
-        socklen_t  ul = setSockAddr(u,returnep,returnep.port);
-        rc = sendto(sock, NULL, 0, 0, &u.sa, ul);
-    }
-    else {
-        rc = send(sock, NULL, 0, SEND_FLAGS);
-    }
-    if (rc < 0) {
+    int rc = wait_read(0);
+    if (rc == 0)
+        return true;
```

Comment:
am a little confused by this condition.
wait_read returns 0 on timeout:
```
        else if (ret == 0)
        {   // timeout
            break;
        }
...
   return ret;
```
that's via poll():
```
       return value of zero indicates that the system call timed out
       before any file descriptors became read.
```

Or is it that when poll is called with timeout=0, and returns 0, it means 0 file descriptors ready not timeout?

If it's correct to treat wait_read call here as not a timeout, then could benefit from a comment describing reasons/semantics.

Reply:
I agree it could do with some comments - I'm struggling to follow the logic.